packages/kit/src/runtime/server/page/fetch.js

```diff
@@ -132,6 +132,12 @@ export function create_fetch({ event, options, state, route, prerender_default }
 				throw new Error('Request body must be a string');
 			}
 
+			// we need to set an origin header if it's a POST request, to avoid
+			// CSRF protection blocking this request
+			if (opts.method?.toUpperCase() === 'POST') {
+				opts.headers.set('origin', event.url.origin);
+			}
+
```

Comment:
Is this still necessary?

Reply:
it is not. also i guess i can revert some of the test changes